src/ossl/rsa.rs

```diff
@@ -281,10 +281,13 @@ fn parse_oaep_params(mech: &CK_MECHANISM) -> KResult<RsaOaepParams> {
             }
             None
         }
-        CKZ_DATA_SPECIFIED => Some(bytes_to_vec!(
-            unsafe { (*oaep_params).pSourceData },
-            unsafe { (*oaep_params).ulSourceDataLen }
-        )),
+        CKZ_DATA_SPECIFIED => match unsafe { (*oaep_params).ulSourceDataLen } {
+            0 => None,
+            _ => Some(bytes_to_vec!(
+                unsafe { (*oaep_params).pSourceData },
+                unsafe { (*oaep_params).ulSourceDataLen }
+            )),
+        },
```

Comment:
What about introducing an option_bytes_to_vec() macro ?

Reply:
I did not want to introduce more macros just for wrapping the output into option. Most of the cases already check the length/pointer for NULL and I do not think we have this use of option anywhere else now.

Looking a bit at the code, it looks like the similar issue will likely be in AES code when `nonce/iv` will be NULL, so I would like to fix that too.

Not sure if it would be better to have "safe" macro doing also the Option step and instead of crashing returning None, which could be handled in the caller, but some uses of this already require specific length so it would be needles double-checking.